backend/data_formatter.py

```diff
@@ -139,11 +140,20 @@ def get_for_display(self) -> tuple[list[dict], list[dict]]:
 
         return display_format
 
+    @staticmethod
+    def _add_back_missing(df: pd.DataFrame) -> pd.DataFrame:
+        df.set_index('date', inplace=True)
+        all_dates = pd.date_range(start=df.index.min(), end=df.index.max(), freq='D')
+        df = df.reindex(all_dates, fill_value=0)
+        df.index = df.index.date
+        df.index.name = 'date'
```

Comment:
For code readability, this should be two separate helpers, one is reindexing the other is adding missing data